src/vs/workbench/contrib/chat/common/promptSyntax/computeAutomaticInstructions.ts

```diff
@@ -110,6 +110,47 @@ export class ComputeAutomaticInstructions {
 		this.sendTelemetry(telemetryEvent);
 	}
 
+	/**
+	 * Checks if any agent instruction files (.github/copilot-instructions.md or agents.md) exist in the workspace.
+	 * Used to determine whether to show the "Generate Agent Instructions" hint.
+	 *
+	 * @returns true if instruction files exist OR if instruction features are disabled (to hide the hint)
+	 */
+	public async hasAgentInstructions(token: CancellationToken): Promise<boolean> {
+		const useCopilotInstructionsFiles = this._configurationService.getValue(PromptsConfig.USE_COPILOT_INSTRUCTION_FILES);
+		const useAgentMd = this._configurationService.getValue(PromptsConfig.USE_AGENT_MD);
+
+		// If both settings are disabled, return true to hide the hint (since the features aren't enabled)
+		if (!useCopilotInstructionsFiles && !useAgentMd) {
+			return true;
+		} const { folders } = this._workspaceService.getWorkspace();
```

Comment:
@digitarald the formatting looks off here as well?

Reply:
Will check, looks like AI editing tool artifacts. I thought husky does check formatting on commit in other repos; might be worth having.